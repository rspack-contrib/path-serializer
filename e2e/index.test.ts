import { createRequire } from 'node:module';
import path from 'node:path';
import { expect, test } from 'vitest';
import { createSnapshotSerializer } from '../dist/esm/index';

const require = createRequire(import.meta.url);

expect.addSnapshotSerializer(
  createSnapshotSerializer({
    cwd: path.resolve(__dirname, '..'),
    workspace: path.resolve(__dirname, '..'),
  }),
);

test('should serialize <ROOT>', () => {
  const root = new URL('.', import.meta.url).href;

  expect(root).toMatchInlineSnapshot(`"file://<ROOT>/e2e/"`);
});

test('should serialize <PNPM_INNER>', () => {
  expect(require.resolve('@rslib/core')).toMatchInlineSnapshot(
    `"<ROOT>/node_modules/<PNPM_INNER>/@rslib/core/dist/index.js"`,
  );
});

test('should serialize Object', () => {
  class MyObj {
    attr: string;
    constructor(attr: string) {
      this.attr = attr;
    }
  }

  const obj = new MyObj(require.resolve('@rslib/core'));
  expect(obj).toMatchInlineSnapshot(`
    MyObj {
      "attr": "<ROOT>/node_modules/<PNPM_INNER>/@rslib/core/dist/index.js",
    }
  `);
});

test('should serialize file content', () => {
  const fileContent = `"use strict";
const a = ${require.resolve('@rslib/core')};
`;

  expect(fileContent).toMatchInlineSnapshot(`
    "\\"use strict\\";
    const a = <ROOT>/node_modules/<PNPM_INNER>/@rslib/core/dist/index.js;
    "
  `);

  expect(fileContent).toMatchSnapshot();
});
