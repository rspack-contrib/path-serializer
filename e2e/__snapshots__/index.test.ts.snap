// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should serialize file content 2`] = `
"\\"use strict\\";
  const a = <ROOT>/node_modules/<PNPM_INNER>/@rslib/core/dist/index.js;
"
`;
